File: Selenium/Project/LMS_Activity7.java
package project;


	import org.openqa.selenium.By;
	import org.openqa.selenium.WebDriver;
	import org.openqa.selenium.WebElement;
	import org.openqa.selenium.chrome.ChromeDriver;
	import org.openqa.selenium.support.ui.WebDriverWait;
	import org.openqa.selenium.support.ui.ExpectedConditions;

import java.time.Duration;
import java.util.List;

	public class LMS_Activity7 {
	    public static void main(String[] args) {
	        
	        System.setProperty("webdriver.chrome.driver", "path/to/chromedriver");

	      
	        WebDriver driver = new ChromeDriver();

	        try {
	           
	            driver.get("https://alchemy.hguy.co/lms");

	            
	            WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
	            wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("nav")));

	          
	            WebElement allCoursesMenuItem = driver.findElement(By.linkText("All Courses"));
	            allCoursesMenuItem.click();

	          
	            wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".ld-course-list")));

	           
	            List<WebElement> courses = driver.findElements(By.cssSelector(".ld-course-list .ld-course"));

	           
	            System.out.println("Number of courses on the page: " + courses.size());
	        } catch (Exception e) {
	            e.printStackTrace();
	        } finally {
	           
	            driver.quit();
	        }
	    }
	}

File: Selenium/Project/lms_Activity7.py

# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait
    # Start the Driver
with webdriver.Firefox() as driver:

    # Declare the wait variable
    wait = WebDriverWait(driver, timeout=10)

      # Navigate to the URL
    driver.get("https://alchemy.hguy.co/lms/")
    driver.maximize_window()

    #Find the menu item that says “All Courses”” and click it
    driver.find_element(By.XPATH, "//ul[contains(@class,'nav-menu')]/li[2]/a[1]").click()
    wait.until(EC.visibility_of_element_located((By.XPATH,"//div[@class='ld-course-list-items row']")))
    courses = driver.find_element(By.XPATH, "//h3[@class='entry-title']")
    print(f'Number of similar elements: {len(courses)}')
    # Close the WebDriver
    driver.quit()
